modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java

```diff
@@ -40,6 +41,20 @@
 public class DocumentMetadataSetResourceImpl
 	extends BaseDocumentMetadataSetResourceImpl {
 
+	@Override
+	public void deleteDocumentMetadataSet(Long documentMetadataSetId)
+		throws Exception {
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPD-32247")) {
+			throw new UnsupportedOperationException();
+		}
+
+		_ddmStructureLinkLocalService.deleteStructureStructureLinks(
+			documentMetadataSetId);
```

Comment:
What will happen if there are documents that contain values for this metadata set? Can you verify that the document is still visible, and that it doesn't try to look for the structure?

Also, shouldn't you clean up the DLFileEntryMetadata entries as well then?

Reply:
Hi @adolfopa, you are correct, we should be removing the DLFileEntryMetadata entries as well. I've pushed another commit to do that. I do have a question about it though, currently we do not have a method to delete DLFileEntryMetadata by structureId, so using the methods available now, I am able to delete the structure, then use _dlFileEntryMetadataLocalService.getNoStructuresFileEntryMetadatas() do retrieve a list of entries that no longer have a structure associated with them.

My question is should this be the approach, or should I add a new method in the service layer to delete by structureId?

> What will happen if there are documents that contain values for this metadata set? Can you verify that the document is still visible, and that it doesn't try to look for the structure?
Regarding this question, yes we can still view the document. The subsection in the information tab for the metadata set is no longer present and no errors are thrown.

Reply:
I think we should add a specific method to the Java API. It should be a matter of simply adding a new finder in the service.xml file.

Reply:
Thanks Adolfo, I've pushed some additional commits to make that change.